Tools/CMake/AMReXInstallHelpers.cmake

```diff
@@ -33,6 +33,13 @@ function ( generate_amrex_config_header )
       set(COMPILER_ID_MACRO  __PGI)
    elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )
       set(COMPILER_ID_MACRO  __llvm__)
+   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" )
+      set(COMPILER_ID_MACRO  __llvm__)
+   else ()
+      # avoid malforming the AMReX_Config.H file by setting something
+      set(COMPILER_ID_MACRO  ${CMAKE_CXX_COMPILER_ID})
```

Comment:
I guess if we fallback to guessing `__llvm__` for new compilers we are 90% of the time right. Shall I set accordingly?